frontend/src/components/Slider/NamedSlider.tsx

```diff
@@ -0,0 +1,82 @@
+import { ReactNode, useState, MouseEventHandler, useRef, KeyboardEventHandler } from 'react';
+
+interface SliderProps {
+  /** children passed to the slider appear in the Slider */
+  children: Array<ReactNode> | ReactNode;
+  /** onChange event is triggered whenever the current percentage value of the slider changes */
+  onChange: (percentage: number) => void;
+}
+
+/**
+ * A Slider with a label which can be controlled by dragging, clicking and arrow up/down.
+ */
+export const NamedSlider = (props: SliderProps) => {
+  const [width, setWidth] = useState(0);
+  const sliderDivRef = useRef<HTMLDivElement>(null);
+
+  const minWidth = 0;
+  const maxWidth = sliderDivRef.current ? sliderDivRef.current.clientWidth : 100;
+
+  const dragHandler: MouseEventHandler = (pointerEvent) => {
+    const startWidth = width;
+    const startPosition = { x: pointerEvent.pageX, y: pointerEvent.pageY };
+
+    function onMouseMove(pointerEvent: PointerEvent) {
+      if (!sliderDivRef.current) return;
+      const w = startWidth - startPosition.x + pointerEvent.pageX;
+      const newWidth = w > minWidth ? (w < maxWidth ? w : maxWidth) : minWidth;
+      const sliderWidth = sliderDivRef.current.clientWidth;
+      setWidth(newWidth);
+      props.onChange(newWidth / sliderWidth);
+    }
+    function onMouseUp() {
+      document.body.removeEventListener('pointermove', onMouseMove);
+    }
+
+    document.body.addEventListener('pointermove', onMouseMove);
+    document.body.addEventListener('pointerup', onMouseUp, { once: true });
+  };
+
+  const clickHandler: MouseEventHandler = (mouseClickEvent) => {
+    if (!sliderDivRef.current) return;
+    const offsetLeft = sliderDivRef.current.getBoundingClientRect().left;
+    const newWidth = mouseClickEvent.pageX - offsetLeft;
+    const sliderWidth = sliderDivRef.current.clientWidth;
+    setWidth(newWidth);
+    props.onChange(newWidth / sliderWidth);
+  };
+
+  const changePercentage = (value: number) => {
+    if (!sliderDivRef.current) return 0;
+    const sliderWidth = sliderDivRef.current.clientWidth;
+    const p = width / sliderWidth + (sliderWidth * value) / sliderWidth;
+    const percentage = p > 1 ? 1 : p < 0 ? 0 : p;
+    setWidth(percentage * sliderWidth);
+    props.onChange(percentage);
+  };
+
+  const keyDownHandler: KeyboardEventHandler = (event) => {
+    if (event.key === 'j' || event.key === 'ArrowUp') {
+      changePercentage(0.05);
+    } else if (event.key === 'k' || event.key === 'ArrowDown') {
```

Comment:
weird choice of keybindings (up/down also scrolls up/down at the same time): what about left+j for reducing, right+k for increasing?

Please also document this. See #324

Reply:
Yes, scrolling is indeed an issue. I could change it to left and right as we don't scroll on the x-axis anyway.
I chose j and k as an alternative because those resemble the vim key bindings for up and down and I personally like when software uses vim keybindings.
All of that were quick decisions to test the feature and are up for debate of course :D

Reply:
> Yes, scrolling is indeed an issue. I could change it to left and right as we don't scroll on the x-axis anyway.

Makes a lot of sense to me :+1: 

There seems to be a bug, though: pressing left increases and right does not work.

> I chose j and k as an alternative because those resemble the vim key bindings for up and down and I personally like when software uses vim keybindings.

Yes, let us use vim keybindings, see #324

Reply:
I messed up the keys :D
Should work now